course/include/isometry.h

```diff
@@ -72,6 +73,8 @@ class Matrix3 {
   Matrix3 operator*(const Matrix3 &other) const;
   Matrix3 operator*(const Vector3 &other) const;
   Matrix3 operator/(const Matrix3 &other) const;
+  Matrix3 product(const Matrix3 &rhs) const;
+  Vector3 product(const Vector3 &rhs) const;
```

Comment:
What are these for? Why not using `operator*()`.

Reply:
The implementation of operator* is not mathematically correct, so had to implement mathematically correct versions of them.

Reply:
You're right. I did not remember `operator*` does a scalar multiplication between matrices.